src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,647 @@ const ActorInit En_Rz_InitVars = {
+        if (animIndex >= ARRAY_COUNT(D_80BFCD20)) {
+            endFrame = Animation_GetLastFrame(sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)]);
+            if (animMode == ANIMMODE_LOOP) {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, transitionRate);
+            } else {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, transitionRate);
+            }
+        } else {
+            Animation_Change(&this->skelAnime, animationPtr[animIndex], 1.0f, 0.0f,
+                             Animation_GetLastFrame(animationPtr[animIndex]), animMode, transitionRate);
+        }
+
+        this->animIndex = animIndex;
+    }
+}
+
+void func_80BFB9E4(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void func_80BFBA1C(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, 0.0f);
+}
+
+s32 func_80BFBA50(EnRz* this, PlayState* play) {
+    if (EN_RZ_GET_PATH(&this->actor) != 0x3F) {
+        this->path = &play->setupPathList[EN_RZ_GET_PATH(&this->actor)];
+        if (this->path != NULL) {
+            Path* path = this->path;
+            Vec3s* points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+            f32 temp_fv0 = points->x - this->actor.world.pos.x;
+            f32 temp_fv1 = points->z - this->actor.world.pos.z;
+
+            if ((SQ(temp_fv0) + SQ(temp_fv1)) < SQ(10.0f)) {
+                this->unk_3BC = 0;
+                this->unk_420 &= ~1;
+            } else {
+                this->unk_3BC = path->count - 1;
+                this->unk_420 |= 1;
+                return true;
+            }
+        }
+    } else {
+        this->path = NULL;
+        this->unk_3BC = 0;
+    }
+    return false;
+}
+
+s32 func_80BFBB44(EnRz* this) {
+    Path* path = this->path;
+    Vec3s* pointPos;
+    f32 temp_fa0;
+    f32 temp_fa1;
+
+    if (path == NULL) {
+        return true;
+    }
+
+    pointPos = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_3BC];
+    temp_fa0 = pointPos->x - this->actor.world.pos.x;
+    temp_fa1 = pointPos->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(temp_fa0, temp_fa1);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 2000, 200);
+
+    if ((SQ(temp_fa0) + SQ(temp_fa1)) < SQ(10.0f)) {
+        if (this->unk_420 & 1) {
+            this->unk_3BC--;
+            if (this->unk_3BC < 0) {
+                return 2;
+            }
+        } else {
+            this->unk_3BC++;
+            if (this->unk_3BC >= path->count) {
+                return 2;
+            }
+        }
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
        return 1;
    }
    return 0;
```
Returns more than just 1 and 0, so doesn't seem boolean